core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -1223,7 +1243,8 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
       }
     }
 
-    done
+    //unreachable
+    ???
```

Comment:
If we're going to go down this path, consider throwing an `IllegalStateException` with a helpful message to help future devs with debugging (I know it's impossible after a `while(true)` but someone will very likely change the condition in the future)